common/lib/xmodule/xmodule/modulestore/inheritance.py

```diff
@@ -231,6 +231,13 @@ class InheritanceMixin(XBlockMixin):
         scope=Scope.settings
     )
 
+    icon = String(
```

Comment:
`InheritanceMixin` is already kind of messy and we're trying not to make that worse. If this new `icon` setting is only for Units, then you should add this field to the `vertical` block directly, to avoid adding it to every other XBlock type when it's not needed.

Reply:
Sure thing, I moved this to the `vertical` block.